service/src/main/java/uk/gov/hmcts/reform/fpl/service/ValidateEmailService.java

```diff
@@ -52,15 +69,74 @@ private boolean emailIsOptional(String key) {
         return key.equals("Applicant");
     }
 
+    /*
+    Mimic gov notify validation
+    see https://github.com/alphagov/notifications-utils/blob/master/notifications_utils/recipients.py#L494-L534
+     */
     public boolean isValid(String email) {
-        InternetAddress internetAddress = new InternetAddress();
-        internetAddress.setAddress(email);
+
+        if (isEmpty(email)) {
+            log.warn("Email is null or empty");
+            return false;
+        }
+
+        final String emailAddress = StringUtils.trim(email);
+
+        if (emailAddress.length() > 320) {
+            log.warn("Email is longer than {} characters", 320);
+            return false;
+        }
+
+        if (emailAddress.contains("..")) {
+            log.warn("Email contains ..");
+            return false;
+        }
+
+        final Matcher emailMatcher = EMAIL_PATTERN.matcher(emailAddress);
+
+        if (!emailMatcher.matches()) {
+            log.warn("Email does not match pattern");
+            return false;
+        }
+
+        String hostname = emailMatcher.group(1);
 
         try {
-            internetAddress.validate();
-            return true;
-        } catch (AddressException addressException) {
+            hostname = toASCII(toUnicode(hostname));
+        } catch (Exception e) {
+            log.warn("Email hostname can not be converted to ascii");
+            return false;
+        }
+
+        final String[] hostParts = hostname.split("\\.");
```

Comment:
Would it be nicer to use `Pattern.quote(".")` instead of the raw string, might make it more readable but don't really mind, if you do it might be worthwhile precomputing it like you do with the patterns.

Reply:
Also it is advised to use a limit of -1 to get consistent output of the method when no explicit limit is required, see https://errorprone.info/bugpattern/StringSplitter